Stylecop fix

Hellhake opened the pull request:
Fix StyleCop warnings

Mailaender commented on OpenRA.Game/FileFormats/Blast.cs:
```diff
@@ -142,9 +142,9 @@ public static byte[] Decompress(byte[] src)
 				else
-				{
+                {
+                    // literal value
```
Please indent with tabs instead of spaces.

Mailaender commented on OpenRA.Game/FileFormats/XccGlobalDatabase.cs:
```diff
@@ -34,7 +34,7 @@ public XccGlobalDatabase(Stream s)
 
 					// Skip comment
-					while ((c = reader.ReadChar()) != 0);
+                    while ((c = reader.ReadChar()) != 0) { }
```
Please indent with spaces rather than spaces.

Mailaender commented on OpenRA.Game/FileSystem/PackageEntry.cs:
```diff
@@ -19,7 +19,8 @@ namespace OpenRA.FileSystem
 	public class PackageEntry
-	{
+    {
+        public const int Size = 12;
```
Spaces again.

Mailaender commented on OpenRA.Game/Game.cs:
```diff
@@ -82,10 +82,7 @@ static void JoinLocal()
-		public const int NetTickScale = 3; // 120 ms net tick for 40 ms local tick
-		public const int Timestep = 40;
-		public const int TimestepJankThreshold = 250; // Don't catch up for delays larger than 250ms
+        public static int LocalTick { get { return orderManager.LocalFrameNumber; } }
```
Spaces...

Mailaender commented on OpenRA.Game/Game.cs:
```diff
@@ -82,10 +82,7 @@ static void JoinLocal()
-		public static int LocalTick { get { return orderManager.LocalFrameNumber; } }
-		public const int NetTickScale = 3; // 120 ms net tick for 40 ms local tick
-		public const int Timestep = 40;
-		public const int TimestepJankThreshold = 250; // Don't catch up for delays larger than 250ms
```
You should not remove those!

Mailaender commented on OpenRA.Game/GameRules/SoundInfo.cs:
```diff
@@ -16,12 +16,12 @@ namespace OpenRA.GameRules
+		[FieldLoader.Ignore] public readonly Dictionary<string, string[]> Voices;
+		[FieldLoader.Ignore] public readonly Dictionary<string, string[]> Notifications;
+		public readonly string DefaultVariant = ".aud"; 
+		public readonly string DefaultPrefix = ""; 
```
Remove the trailing spaces.

Mailaender commented on OpenRA.Game/Input/Keycode.cs:
```diff
@@ -420,7 +420,7 @@ public static class KeycodeExts
 			{ Keycode.CURRENCYSUBUNIT, "CurrencySubUnit" },
 			{ Keycode.KP_LEFTPAREN, "Keypad (" },
-			{ Keycode.KP_RIGHTPAREN, "Keypad )" },
+			{ Keycode.KP_RIGHTPAREN, "Keypad)" },
```
This is most likely a false positive. Revert it please.

Mailaender commented on OpenRA.Game/Map/Map.cs:
```diff
@@ -112,7 +112,8 @@ public enum MapVisibility
 	public class Map
-	{
+    {
+        public const int MaxTilesInCircleRange = 50;
```
Spaces again.

Mailaender commented on OpenRA.Game/Network/UnitOrders.cs:
```diff
@@ -1,4 +1,4 @@
-#region Copyright & License Information
+﻿#region Copyright & License Information
```
Can you avoid committing these changes in general especially if it is the only change in the whole file?

Phrohdoh replied:
This looks to be adding BOM which is a no-no.

Mailaender commented on OpenRA.Game/Selection.cs:
```diff
@@ -64,6 +64,7 @@ public void Tick(World world)
 			actors.RemoveAll(a => !a.IsInWorld || (!a.Owner.IsAlliedWith(world.RenderPlayer) && world.FogObscures(a)));
 
 			foreach (var cg in controlGroups.Values)
+
```
This is weird now. Probably just add curly braces instead.

Mailaender commented on OpenRA.Game/Server/Connection.cs:
```diff
@@ -16,14 +16,14 @@
 	public class Connection
-	{
+    {
+        public const int MaxOrderLength = 131072;
```
You should change your IDE project settings to use tabs instead of spaces for OpenRA. We support http://editorconfig.org/

Mailaender commented on OpenRA.Game/Support/PerfTimer.cs:
```diff
@@ -17,7 +17,10 @@
+    {
+        const int Digits = 6;
+        const string IndentationString = "|   ";
+        const string FormatSeperation = " ms ";
```
Spaces instead of tabs seems to happen every time you move a line.

Mailaender commented on OpenRA.Game/Support/Program.cs:
```diff
@@ -86,8 +86,9 @@ static StringBuilder BuildExceptionReport(Exception e, StringBuilder sb, int d)
-			else // TODO: more exception types
+			else
 			{
+                // TODO: more exception types
```
Tabs please.

Mailaender commented on OpenRA.Game/Traits/Player/PlayerResources.cs:
```diff
@@ -23,7 +23,9 @@ public class PlayerResourcesInfo : ITraitInfo
-	{
+    {
+        const float displayCashFracPerFrame = .07f;
+        const int displayCashDeltaPerFrame = 37;
```
Tabs please.

Mailaender commented on OpenRA.Game/Widgets/ButtonWidget.cs:
```diff
@@ -149,14 +149,15 @@ public override bool HandleMouseInput(MouseInput mi)
 			else if (HasMouseFocus && mi.Event == MouseInputEvent.Up)
-			{
+            {
+                // Only fire the onMouseUp event if we successfully lost focus, and were pressed
```
Tabs please.

Mailaender commented on OpenRA.Game/World.cs:
```diff
@@ -116,7 +116,11 @@ public void SetLocalPlayer(string pr)
+		    get
+		    {
+		        return orderGenerator_;
+		    }
```
A horrible mix of spaces and tabs.

Hellhake replied:
Auto generated by Resharper ;-)

Mailaender commented on OpenRA.Mods.Common/Graphics/VoxelRenderable.cs:
```diff
@@ -1,4 +1,4 @@
-#region Copyright & License Information
+﻿#region Copyright & License Information
```
Please omit those changes. They will most likely conflict with https://github.com/OpenRA/OpenRA/pull/7173.

Mailaender commented on OpenRA.Mods.Common/LoadScreens/ModChooserLoadScreen.cs:
```diff
@@ -50,4 +50,4 @@ public void Dispose()
 	}
-}
\ No newline at end of file
+}
```
Same here, nothing changed in this file.

Mailaender commented on OpenRA.Mods.Common/Widgets/LabelWithTooltipWidget.cs:
```diff
@@ -1,4 +1,4 @@
-#region Copyright & License Information
+﻿#region Copyright & License Information
```
As a rule of thumb. The whole OpenRA.Mods.Common is already StyleCop clean so please don't commit any changes to that directory.

Mailaender commented on OpenRA.Mods.Common/Widgets/Logic/SettingsLogic.cs:
```diff
@@ -350,18 +350,14 @@ Action InitInputPanel(Widget panel)
 					{ "NextProductionTabKey", "Next production tab" },
 					{ "PreviousProductionTabKey", "Previous production tab" },
 					{ "CycleProductionBuildingsKey", "Cycle production facilities" },
-
```
Please revert this. We added an exception here. https://github.com/OpenRA/OpenRA/pull/7193#discussion_r22280295

Mailaender commented on OpenRA.Mods.RA/AI/AttackOrFleeFuzzy.cs:
```diff
@@ -14,6 +14,7 @@
 using AI.Fuzzy.Library;
 using OpenRA.Mods.RA.Traits;
 using OpenRA.GameRules;
+using OpenRA.Mods.RA.Move;
```
This is probably a merge conflict edited towards the wrong direction. Remove it again.

Mailaender commented on OpenRA.Mods.D2k/Widgets/SupportPowerBinWidget.cs:
```diff
@@ -181,4 +181,4 @@ static Action<MouseInput> HandleSupportPower(string key, SupportPowerManager man
 	}
-}
\ No newline at end of file
+}
```
OpenRA.Mods.D2k is also already StyleCop clean. Remove those non-changes.

Mailaender commented on OpenRA.Mods.RA/AI/BaseBuilder.cs:
```diff
@@ -73,10 +73,9 @@ bool TickQueue(ProductionQueue queue)
 			else if (currentBuilding.Done)
-			{
+            {
+                // Production is complete
```
The usual spaces instead of tabs problem.

Mailaender commented on OpenRA.Mods.RA/Activities/DeliverResources.cs:
```diff
@@ -18,11 +18,11 @@
 	public class DeliverResources : Activity
-	{
+    {
+        const int NextChooseTime = 100;
```
:(

Mailaender commented on OpenRA.Mods.RA/Activities/LayMines.cs:
```diff
@@ -56,16 +56,16 @@ public override Activity Tick(Actor self)
 			if (ml.Minefield.Length > 0)
 			{
-				for (var n = 0; n < 20; n++)		// dont get stuck forever here
+                // dont get stuck forever here
```
Tabs please.

Mailaender commented on OpenRA.Mods.RA/Activities/Sell.cs:
```diff
@@ -11,7 +11,7 @@
 using OpenRA.Mods.Common.Effects;
 using OpenRA.Mods.Common.Traits;
-using OpenRA.Mods.RA.Traits;
+using OpenRA.Mods.RA.Buildings;
```
That looks spurious.

Mailaender commented on OpenRA.Mods.RA/Activities/Teleport.cs:
```diff
@@ -29,7 +29,6 @@ public class Teleport : Activity
 		bool screenFlash;
 		string sound;
 
-		const int maxCellSearchRange = Map.MaxTilesInCircleRange;
```
You probably shouldn't just remove this.

Mailaender commented on OpenRA.Mods.RA/CombatDebugOverlay.cs:
```diff
@@ -10,6 +10,7 @@
 
 using System;
 using System.Drawing;
+
```
Spurious newline.

Mailaender commented on OpenRA.Mods.RA/Player/AllyRepair.cs:
```diff
@@ -10,8 +10,7 @@
 
-namespace OpenRA.Mods.RA.Traits
-
+namespace OpenRA.Mods.RA
```
Please revert this revert.

Mailaender commented on OpenRA.Mods.RA/ProximityCapturable.cs:
```diff
@@ -70,7 +70,8 @@ public void Tick(Actor self)
 				var captor = GetInRange(self);
 
-				if (captor != null) // got one
+                // got one
```
Tabs

Mailaender commented on OpenRA.Mods.RA/SupportPowers/SupportPowerManager.cs:
```diff
@@ -189,7 +189,8 @@ public void Tick()
 
-				if (RemainingTime > 0) --RemainingTime;
+				if (RemainingTime > 0)
+                    --RemainingTime;
```
Tabs and probably a newline.

Mailaender commented on OpenRA.Mods.RA/TakeCover.cs:
```diff
@@ -46,7 +46,8 @@ public TakeCover(ActorInitializer init, TakeCoverInfo info)
 		public void Damaged(Actor self, AttackInfo e)
 		{
-			if (e.Damage > 0 && (e.Warhead == null || !e.Warhead.PreventProne)) /* Don't go prone when healed */
+            /* Don't go prone when healed */
```
Tabs

Mailaender commented on OpenRA.Mods.RA/Traits/World/PathFinder.cs:
```diff
@@ -26,8 +26,9 @@ public class PathFinderInfo : ITraitInfo
-	{
-		readonly static List<CPos> emptyPath = new List<CPos>(0);
+    {
+        const int MaxPathAge = 50;	/* x 40ms ticks */
```
Tabs please.

LipkeGu replied:
is an VS settings problem...

Mailaender commented:
If a whole project folder is StyleCop clean after this massive cleanup you should add them to https://github.com/OpenRA/OpenRA/blob/bleed/Makefile#L193 so our CI checks pull request against new violations.

Phrohdoh commented:
I don't think it is necessary to have a commit for each fix.

Mailaender commented:
That will also make doing your fixes with `git amend` a lot easier.

Hellhake committed:
Remove BOM

LipkeGu commented:
@Hellhake: When you are on Windows and you are using Visual Studio then you should and YOU HAVE TO change a few settings. Please got to
```
Extras -> Options -> (left in the Tree) Text Editor -> All languages -> tabulators -> keep tabulators
```
also you have to setup git to convert all line endings to ```Unix``` when pushing/commiting:
``` checkout as is, convert to unix line endings when commiting / pushing```

LipkeGu commented on OpenRA.Game/Graphics/SpriteFont.cs:
```diff
@@ -1,4 +1,4 @@
-#region Copyright & License Information
+﻿#region Copyright & License Information
```
line ending issue

LipkeGu commented on OpenRA.Game/Graphics/SpriteRenderer.cs:
```diff
@@ -1,4 +1,4 @@
-#region Copyright & License Information
+﻿#region Copyright & License Information
```
Check the line ending...

Hellhake committed:
Fix StyleCop warnings in OpenRA.Editor

Hellhake committed:
Fix StyleCop warnings in OpenRA.Game

Hellhake committed:
Fix StyleCop warnings in OpenRA.Mods.RA

Hellhake committed:
Fix StyleCop warnings in OpenRA.Test

Hellhake committed:
Fix StyleCop warnings in OpenRA.Mods.Common

Hellhake committed:
Fix StyleCop warnings in OpenRA.Utility

Hellhake committed:
Fix StyleCop warnings in OpenRA.Renderer.Sdl2

Hellhake commented:
Ok, i think all issues should now be resolved.
Additionally i fixed the stylecop warnings which where created after my fixes and squashed everything into one single commit per Project.

Hellhake commented:
@LipkeGu
I'm using Linux as well as Windows.
On Windows i told Git to "Checkout Windows-style, commit Unix-style line endings".
I think this should be fine.

Thanks for the tip with the Visual Studio setting but i usually prefer space indentation.
Is their a project setting for this?

Hellhake commented:
Sorry about the gigantic indentation.
I just converted spaces to tabs.
And these lines apparently contained 11 * 3 spaces.
These were converted to 11 tabs.
